workspace/commands/helpers.py

```diff
@@ -60,17 +60,23 @@ def envsection(self, env=None):
 
     @property
     def workdir(self):
-        return os.path.join(self.path, self.get('tox', 'toxworkdir', '.tox'))
+        toxworkdir = self.get('tox', 'toxworkdir', '.tox')
+        toxworkdir = toxworkdir.replace('{homedir}', self.homedir)
+        toxworkdir = toxworkdir.replace('{toxinidir}', self.path)
+        return os.path.join(self.path, toxworkdir)
 
     @property
     def homedir(self):
         return os.path.expanduser('~')
 
     def envdir(self, env):
-        default_envdir = os.path.join(self.path, '.tox', env)
-        return self.expand_vars(self.get(self.envsection(env), 'envdir', self.get(self.envsection(),
-                                         'envdir', default_envdir)).replace('{toxworkdir}', self.workdir),
-                                {'envname': env})
```

Comment:
let's add `self.toxworkdir` property alias to `self.workdir`, then everything should just work via `expand_vars`, and there should be no need to change this line.